src/optimization/inlineConstructors.ml

```diff
@@ -42,47 +41,69 @@ open Globals
 	inlining.
 
 	Second pass:
-	Replace variables that alias inline objects with their respective field inline variables.
-	Identify inline objects by order of appearance and replace them with their inlined constructor expressions.
-	Replace field access of aliasing variables with the respective field inline variable.
+	Replaces inline objects with their inlined constructor expressions.
+	Replaces field access of aliasing variables with the respective field inline variable or inlined methods.
 	Because some replacements turn a single expression into many, this pass will map texpr into texpr list,
 	which is converted into TBlocks by the caller as needed.
 *)
 
 type inline_object_kind =
-	| IOKCtor of tclass_field * bool * tvar list
+	| IOKCtor of
+		tclass *
+		Type.tparams *
+		tclass_field *
+		bool (* Wether the inline constructor is forced or not. Cancelling a forced constructor should produce an error *)
```

Comment:
I recommend using a record type instead of a tuple type here, in my experience that's easier to work with and also easier to modify.

Reply:
I've committed this change.